src/Search/Queue/WatchableQueue.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+namespace eLife\Search\Queue;
+
+interface WatchableQueue
+{
+    /**
+     * Adds item to the queue.
+     *
+     * Mock: Add item to queue.
+     * SQS: This will set the queue item into the memory slot for re-processing.
+     */
+    public function enqueue(QueueItem $item) : bool;
+
+    /**
+     * Starts process of removing item.
+     *
+     * Mock: Move to separate "in progress" queue.
+     * SQS: this will change the timeout of the in-memory item.
+     */
```

Comment:
I guess the `timeout` is the visibility timeout, makes sense to be able to override it if there is a particularly heavy message to process, but given the queue is just one this could also go into the configuration of the `WatchableQueue` object to make the interface simpler.